### PR TITLE
chore!: turn on Versioned Export in config.py

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY

This PR turns the Versioned Export flag to true, this is a breaking change for superset 2.0

### ADDITIONAL INFORMATION

- [ ] Has associated issue: https://github.com/apache/superset/issues/18688
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: UPDATING.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ assists people when migrating to a new version.
 - [18970](https://github.com/apache/superset/pull/18970): Changes feature
 flag for the legacy datasource editor (DISABLE_LEGACY_DATASOURCE_EDITOR) in config.py to True, thus disabling the feature from being shown in the client.
 - [19017](https://github.com/apache/superset/pull/19017): Removes Python 3.7 support.
+- [19142](https://github.com/apache/superset/pull/19142): Changes feature flag for versioned export(VERSIONED_EXPORT) to be true, depreciating older api endpoints.
```
Comment (betodealmeida) [inline UPDATING.md:40]:
```suggestion
- [19142](https://github.com/apache/superset/pull/19142): Changes feature flag for versioned export (VERSIONED_EXPORT) to be true, deprecating older api endpoints.
```

File: superset/config.py [hunk 2]
```diff
@@ -407,7 +407,7 @@ def _try_json_readsha(filepath: str, length: int) -> Optional[str]:
     "DASHBOARD_NATIVE_FILTERS_SET": False,
     "DASHBOARD_FILTERS_EXPERIMENTAL": False,
     "GLOBAL_ASYNC_QUERIES": False,
-    "VERSIONED_EXPORT": False,
+    "VERSIONED_EXPORT": True,
```
Comment (betodealmeida) [inline superset/config.py:410]: ❤️

**Review by betodealmeida — APPROVED** [review]

File: tests/integration_tests/cli_tests.py [hunk 3]
```diff
@@ -335,111 +283,6 @@ def test_import_datasets_versioned_export(import_datasets_command, app_context,
     import_datasets_command.assert_called_with(expected_contents, overwrite=True)
 
 
-@mock.patch.dict(
-    "superset.config.DEFAULT_FEATURE_FLAGS", {"VERSIONED_EXPORT": False}, clear=True
```
Comment (betodealmeida) [inline tests/integration_tests/cli_tests.py:339]: Try `superset.cli.lib.feature_flag` instead of `superset.config.DEFAULT_FEATURE_FLAGS`

**Review by betodealmeida — CHANGES_REQUESTED** [review]
Talked offline, let's keep the unit tests.

File: tests/integration_tests/dashboards/api_tests.py [hunk 4]
```diff
@@ -1346,7 +1346,9 @@ def test_export(self):
         # freeze time to ensure filename is deterministic
         with freeze_time("2020-01-01T00:00:00Z"):
             rv = self.get_assert_metric(uri, "export")
-            headers = generate_download_headers("json")["Content-Disposition"]
+            headers = generate_download_headers(
```
Comment (betodealmeida) [inline tests/integration_tests/dashboards/api_tests.py:1349]: Can you leave this test unmodified and make it run with the feature flag off?

```python
    @patch.dict(
        "superset.extensions.feature_flag_manager._feature_flags",
        {"VERSIONED_EXPORT": False},
        clear=True,
)
```

And maybe remove the feature flag patch from the `test_export_bundle` test below on line 1406.

File: UPDATING.md (hunk 1 above)
Comment (betodealmeida) [reply UPDATING.md:40]: This is still unresolved. :)

File: tests/integration_tests/dashboards/api_tests.py (hunk 4 above)
Comment (AAfghahi) [reply tests/integration_tests/dashboards/api_tests.py]: Went through and removed this feature flag patch when set to true for a couple of tests.

**Review by betodealmeida — APPROVED** [review]

**Pull request merged**